www/src/theme/BlogPostPage/Metadata/index.tsx

```diff
@@ -0,0 +1,71 @@
+import { PageMetadata } from '@docusaurus/theme-common';
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+import { useBlogPost as ubp } from '@docusaurus/theme-common/internal';
+import { useBlogPost as typedUbp } from '@docusaurus/theme-common/lib/internal';
+import useDocusaurusContext from '@docusaurus/useDocusaurusContext';
+import React from 'react';
+import { searchParams } from '../../../pages';
+
+// /lib doesn't actually export but actual export has import error in vscode
+const useBlogPost = ubp as unknown as typeof typedUbp;
+
+export default function BlogPostPageMetadata(): JSX.Element {
+  const { assets, metadata } = useBlogPost();
+  const { title, description, date, tags, authors, frontMatter } = metadata;
+  const { keywords } = frontMatter;
+  const _image = assets.image ?? frontMatter.image;
+  const {
+    siteConfig: { customFields },
+  } = useDocusaurusContext();
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const url = customFields!['url']! as string;
+
+  console.log({ metadata });
+
+  const author = authors[0];
+
+  const ogImg = `https://www-git-og-images-trpc.vercel.app/api/ogBlog?${searchParams(
+    {
+      title: metadata.title,
+      description: metadata.description,
+      authorName: author.name as string,
+      authorTitle: author.title as string,
+      authorImg: author.imageURL as string,
+      date,
+      readingTime: (metadata.readingTime as number).toString(),
+    },
```

Comment:
This could use the actual `typeof ogBlogSchema['_input']`

```ts
const params: typeof ogBlogSchema['_input'] = {
  // yay, type-safe and autocompletion
}

const ogImg = `https://www-git-og-images-trpc.vercel.app/api/ogBlog?${searchParams(input)}`

Reply:
Alternatively `searchParams` could take a generic with the shape

```ts
const ogImg = `https://www-git-og-images-trpc.vercel.app/api/ogBlog?${searchParams<typeof ogBlogSchema['_input']>({
  // yay, autocompletion
})}
```